src/devices/bus/nes_ctrl/turbofile.h

```diff
@@ -0,0 +1,57 @@
+// license:BSD-3-Clause
+// copyright-holders:kmg
+/**********************************************************************
+
+    Nintendo Family Computer ASCII Turbofile SRAM external storage device
+
+**********************************************************************/
+
+#ifndef MAME_BUS_NES_CTRL_TURBOFILE_H
+#define MAME_BUS_NES_CTRL_TURBOFILE_H
+
+#pragma once
+
+#include "ctrl.h"
+#include "machine/nvram.h"
+
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> nes_turbofile_device
+
+class nes_turbofile_device : public device_t,
+							public device_nes_control_port_interface
+{
+public:
+	// construction/destruction
+	nes_turbofile_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	virtual ioport_constructor device_input_ports() const override;
```

Comment:
Isn’t this protected in the base class?  It should need to be lifted to public.

Reply:
I see. They're all over the place in derived classes. Fixed.